src/matrix.rs

```diff
@@ -161,6 +161,39 @@ pub fn send_message(
 	handle.perform().or_else(error::map_curl_error)
 }
 
+pub fn send_html_message(
+	homeserver: &str,
+	access_token: &str,
+	room_id: &str,
+	body: &str,
+) -> Result<()> {
+	let mut handle = Easy::new();
+	handle
+		.url(
+			format!(
+				"{}/_matrix/client/r0/rooms/{}/send/m.room.message?access_token={}",
```

Comment:
Building URLs via `format!` seems a bit shady to me. Especially when embedding access tokens some URL library is likely a better choice, no?

Reply:
Agreed. In fact I had a `FIXME` about that somewhere and tried to switch the code to [this library](https://docs.rs/url/2.1.1/url/), but had some issues. Unfortunately I don't remember what they were any more. I'll give it another go.